### ISSUE TITLE
Support for external clusters?

### ISSUE THREAD

**Issue by nabuskey** [cnoe-io/idpbuilder#74]
Should we support external clusters as the target cluster to install packages? 

Dependencies? 

Be able to pass references to existing clusters? 

Not creating external clusters? 

Have two go modules in the repo to avoid creating hard dependencies for local stuff. 

We could target VCluster for easier bootstraping things the same way we are doing for local kind cluster.

For release v1.0.0

**Comment by cmoulliard** [comment]
> One use case would be:

Another user story would be:

As an IDP builder, I would like to provision an existing cluster with the needed packages (core, pluggable) to set for a team, organisation their IDP 

**Note**: This user story is equivalent to what idpbuilder is doing today except that it will not bump a kind cluster + docker registry BUT instead will deploy: gitea, argocd, cert manager (= core packages) and pluggable (= what it should be deployed using ArgoCD Application CRs + config (= Hooks)

**Comment by csantanapr** [comment]
I strongly agree with the user story @cmoulliard shared
>As an IDP builder, I would like to provision an existing cluster with the needed packages (core, pluggable) to set for a team, organisation their IDP
Note: This user story is equivalent to what idpbuilder is doing today except that it will not bump a kind cluster + docker registry BUT instead will deploy: gitea, argocd, cert manager (= core packages) and pluggable (= what it should be deployed using ArgoCD Application CRs + config (= Hooks)

**Comment by nimakaviani** [comment]
+1 to what @cmoulliard suggested.

you should be able to replicate what you create with the kind cluster on a remote cluster (vCluster or otherwise) with very little effort.

**Comment by blakeromano** [comment]
I know there has been talks of CNOE adoption a GitOps Bridge style approach. I wonder how we much we can lean into the patterns that GitOps Bridge has with ApplicationSets, overrides based on environment, cluster, etc; and be able to pull that into when we get the ability to add packages to remote clusters....

Ideally we'd have a command or something that would say "Based on your GitOps Bridge repo location, and what do you want to deploy we add these files into your directory to add that package"... I think the main goal I'd want to emphasize is that IDP Builder should be interacting with Git on your local machine, maybe Git Provider eventually (Github, GitLab, Gitea, etc;) not your cluster directly.

**Comment by cmoulliard** [comment]
> I think the main goal I'd want to emphasize is that IDP Builder should be interacting with Git on your local machine, maybe Git Provider eventually (Github, GitLab, Gitea, etc;) not your cluster directly.

I agree with the fact that idpbuilder should not become a ClusterManager platform. Anyway, in a future release, we should have the option to request to provision an external cluster and not the internal one if we pass as parameter the kubeconfig (= where context, user, server url are defined) to provision it. Of course, that will be the responsibility of the platform engineer to provide enough RBAC to the user connected to the cluster to provision the platform using ArgoCD + "resources" pushed on the idpbuilder's git repository (or an external one if idpbuilder supports also in the future to opt for an external one)

**Comment by nimakaviani** [comment]
> ideally we'd have a command or something that would say "Based on your GitOps Bridge repo location, and what do you want to deploy we add these files into your directory to add that package"... I think the main goal I'd want to emphasize is that IDP Builder should be interacting with Git on your local machine, maybe Git Provider eventually (Github, GitLab, Gitea, etc;) not your cluster directly.

I wonder whether this should be achieved by combining backstage templates and workflow orchestrators that push the added configs to gitea.
